### PR TITLE
[cssom-view] Remove `caret range` concept from `CaretPosition` interface.

### PR DESCRIPTION
**Pull request by siliu1**
Per #10230, `CaretPosition` backed by live range can cause issues. We should remove the `caret range` concept from `CaretPosition` interface.

This is also a long-standing issue. https://lists.w3.org/Archives/Public/www-style/2011May/0051.html

### CODE REVIEW

File: cssom-view-1/Overview.bs [hunk 1]
```diff
@@ -1118,7 +1114,7 @@ in that the latter can return a <code>frameset</code> element.
 <h3 id=the-caretposition-interface>The {{CaretPosition}} Interface</h3>
 
 A <dfn>caret position</dfn> gives the position of a text insertion point indicator. It always has an associated
-<dfn>caret node</dfn>, <dfn>caret offset</dfn>, and <dfn>caret range</dfn>. It is represented by a {{CaretPosition}} object.
+<dfn>caret node</dfn>, and <dfn>caret offset</dfn>. It is represented by a {{CaretPosition}} object.
```
Comment (kbabbitt) [inline cssom-view-1/Overview.bs:1117]:
```suggestion
<dfn>caret node</dfn> and <dfn>caret offset</dfn>. It is represented by a {{CaretPosition}} object.
```

File: cssom-view-1/Overview.bs [hunk 2]
```diff
@@ -1136,20 +1132,17 @@ The <dfn attribute for=CaretPosition>offset</dfn> attribute must return the <a>c
...
 1. If <a>caret node</a> is a text entry widget that is a replaced element,
     and that is in the document,
     return a [=scaled=] {{DOMRect}} object for the caret in the widget
     as represented by the <a>caret offset</a> value.
     The <a>transforms</a> that apply to the element and its ancestors are applied.
-1. Return null.
+1. Otherwise:
 
-Note: This {{DOMRect}} object is not <a spec=html>live</a>.
+    1. Let <var>caretRange</var> be a collapsed {{Range}} object which [=range/start node=] and [=range/end node=] are set to <a>caret node</a>, and which [=range/start offset=] and [=range/end offset=] are set to <a>caret offset</a>.
```
Comment (kbabbitt) [inline cssom-view-1/Overview.bs:1142]:
```suggestion
    1. Let <var>caretRange</var> be a collapsed {{Range}} object whose [=range/start node=] and [=range/end node=] are set to <a>caret node</a>, and whose [=range/start offset=] and [=range/end offset=] are set to <a>caret offset</a>.
```

**Review by dandclark — APPROVED** [review]

**Review by kbabbitt — APPROVED** [review]

File: cssom-view-1/Overview.bs [hunk 3]
```diff
@@ -1136,20 +1132,17 @@ The <dfn attribute for=CaretPosition>offset</dfn> attribute must return the <a>c
...
     The <a>transforms</a> that apply to the element and its ancestors are applied.
-1. Return null.
+1. Otherwise:
 
-Note: This {{DOMRect}} object is not <a spec=html>live</a>.
+    1. Let <var>caretRange</var> be a collapsed {{Range}} object whose [=range/start node=] and [=range/end node=] are set to <a>caret node</a>, and whose [=range/start offset=] and [=range/end offset=] are set to <a>caret offset</a>.
+    1. Return the {{DOMRect}} object which is the result of invoking the {{Range/getBoundingClientRect()}} method on <var>caretRange</var>.
 
-Issue(10230): Consider removing <a>caret range</a> concept from <a>caret position</a> interface.
+Note: This {{DOMRect}} object is not <a spec=html>live</a>.
```
Comment (sanketj) [inline cssom-view-1/Overview.bs:1145]: Should we expand on this and clarify that the `CaretPosition` itself, not just the `DOMRect` is not live?

Comment (zcorpan) [reply cssom-view-1/Overview.bs:1145]: Sure.

**Review by zcorpan — APPROVED** [review]

File: cssom-view-1/Overview.bs (hunk 3 above)
Comment (siliu1) [reply cssom-view-1/Overview.bs:1147]: Done.

File: cssom-view-1/Overview.bs [hunk 4]
```diff
@@ -1136,20 +1134,17 @@ The <dfn attribute for=CaretPosition>offset</dfn> attribute must return the <a>c
...
 1. If <a>caret node</a> is a text entry widget that is a replaced element,
     and that is in the document,
     return a [=scaled=] {{DOMRect}} object for the caret in the widget
     as represented by the <a>caret offset</a> value.
     The <a>transforms</a> that apply to the element and its ancestors are applied.
-1. Return null.
+1. Otherwise:
 
-Note: This {{DOMRect}} object is not <a spec=html>live</a>.
+    1. Let <var>caretRange</var> be a collapsed {{Range}} object whose [=range/start node=] and [=range/end node=] are set to <a>caret node</a>, and whose [=range/start offset=] and [=range/end offset=] are set to <a>caret offset</a>.
```
Comment (sanketj) [inline cssom-view-1/Overview.bs:1144]: This seems okay for now given this is how both Firefox and Chrome/Edge implement this. However, it seems less than ideal to have to create a range in order to get a bounding client rect. That range will still need to be maintained across mutations until it is garbage collected, which may not happen immediately. Is there a different option we can pursue that doesn't use a live range at all?

**Review by sanketj — APPROVED** [review]

**Review by songjianet — APPROVED** [review]

File: cssom-view-1/Overview.bs (hunk 4 above)
Comment (siliu1) [reply cssom-view-1/Overview.bs:1144]: Thanks. I am only aware of such way to get DOMRect from (node, offset). Totally agree that we should update the spec if we have better way to do that.

**Pull request merged**